frontend/class-twitter.php

```diff
@@ -351,6 +351,13 @@ private function get_twitter_id( $id ) {
 	 */
 	protected function image() {
 
+		/**
+		 * Filter: wpseo_add_twitter_images - Allow developers to add images to the Twitter tags
+		 *
+		 * @api WPSEO_Twitter The current object.
+		 */
+		do_action( 'wpseo_add_twitter_images', $this );
```

Comment:
Doing this action here is too open, because it gives the possibility to add images for singular items as well. 

I would suggest to extends the method: `taxonomy_image_output`, see:
![add_action_to_allow_developers_to_add_images_to_the_twitter_tags_by_r33d3m33r_ _pull_request__8049_ _yoast_wordpress-seo](https://user-images.githubusercontent.com/325040/31711465-857f37ce-b3f8-11e7-9f54-45f8bc720b0c.png)

```
       /**
		 * Filter: wpseo_twitter_taxonomy_image - Allow developers to set a image for the taxonomy.
         *
 		 * @api bool False by default, for not having a taxonomy image
         * 
         * @returns false|string A string when an image should be outputted. False if not.
 		 */
        $img = apply_filters( 'wpseo_twitter_taxonomy_image', false );
        if ( is_string( $img ) && ! empty( $img ) ) {
            $this->image_output( $img );
            return true;
        }
```

If there is not an image set for the taxonomy the filter will be applied and that will solve the situation you are mentioning.